add basic code

MaksimCharyev opened the pull request:
Код взаимодействия с базой данных и nats. Также присутствует примитивный SQL код для создания базы данных

MaksimCharyev committed:
add basic code

Код взаимодействия с базой данных и  nats. Также присутствует примитивный SQL код для создания базы данных

MaksimCharyev committed:
Update db.go

Недостающий код

te32012 commented on db/internal/db/db.go:
```diff
@@ -0,0 +1,49 @@
+)
+
+type DBConnection struct {
+	Conn *pgx.Conn
```
Стоит использовать  пул соединений - иначе постоянно будет блокироваться т к пг только 1 соедиение на поток может поддерживать

te32012 commented on db/internal/db/db.go:
```diff
@@ -0,0 +1,49 @@
+}
+
+func ConnectDB() (DBConnection, error) {
+	conn, err := pgx.Connect(context.Background(), "postgres://username:password@localhost:5432/mydb")
```
Стоит сделать подгрузку из файла

te32012 commented on db/internal/db/db.go:
```diff
@@ -0,0 +1,49 @@
+}
+
+// Сохранение данных студента в базе данных
+func (db DBConnection) SaveStudent(student models.Student) error {
```
контекст передается из внешнего метода

te32012 reviewed (changes requested):
надо исправить то, что указано

te32012 commented on db/internal/db/DB.sql:
Стоит  изменить varchar на тест - поскольку текст автоматический размер выставляет

te32012 commented on db/internal/models/entity.go:
Стоит класс студента = брать там где он передается в натс 
один класс для всех проектов - так его проще будет менять

te32012 commented on db/internal/nats/nats.go:
Метод .nextMessange() стоит заменить конструкцией for msg := range sub (пример)

MaksimCharyev committed:
review changes

db.go changed, nats.go changed, sql file changed. Dockerfile added

te32012 commented on db/cmd/main.go:
```diff
@@ -0,0 +1,22 @@
+	"log"
+)
+
+func main() {
```
стоит все содержание метода вынести в app класс метод Run и дергать его

te32012 commented on db/Dockerfile:
```diff
@@ -0,0 +1,20 @@
+# Указываем базовый образ
+FROM golang:1.20-alpine
```
если работает - ок
но я перепишу

te32012 reviewed (changes requested):
поменяй плез

Commit:
Add ui

Commit:
MVP done

MaksimCharyev committed:
New changes

Commit:
Mobile version done

Commit:
Fix dropdown

Commit:
Dropdown done

Commit:
Update readme

Commit:
Fix bug with dropdown

Commit:
Fix router

mist-developer committed:
Add struct to document processor

mist-developer committed:
Add final changes for template to work

mist-developer committed:
Add Viper support
maybe now it works with env vars

te32012 committed:
начата подготовка к развертыванию

te32012 committed:
начата подготовка к развертыванию (#6)

* Add ui

* MVP done

* Mobile version done

* Fix dropdown

* Dropdown done

* Update readme

* Fix bug with dropdown

* Fix router

---------

Co-authored-by: ilmarch1 <ilmarch1@mts.ru>

Commit:
Fix error

te32012 committed:
Merge branch 'deployment' into feature/ui

te32012 committed:
Feature/UI (#8)

* Add ui

* MVP done

* Mobile version done

* Fix dropdown

* Dropdown done

* Update readme

* Fix bug with dropdown

* Fix router

* Fix error

---------

Co-authored-by: ilmarch1 <ilmarch1@mts.ru>

te32012 committed:
композ без джинкса

Commit:
Fix nginx conf

Commit:
Merge branch 'main' of https://github.com/mpu-project-dpo/documentsGenerantorAll into feature/ui

Commit:
Fix

Commit:
Fix error with tsc

MaksimCharyev committed:
New entity architecture

- Database is reduced to 3NF
- Added Passport entity
- Added University entity
- Added new entity saving methods to db.go
- Added Docker & docker-compose files
- Updated sql script

te32012 committed:
прототип для yandex cloud

Commit:
fix docker

Commit:
add env placeholder

te32012 committed:
Merge branch 'deployment' into fix/fix-deploy

te32012 committed:
Merge pull request #9 from mpu-project-dpo/fix/fix-deploy

Fix/fix deploy

MaksimCharyev committed:
Improved code for db.go

- Now using nats-contracts entity document
- Changed methods to save entities to db
- Using transactions in db.go

te32012 committed:
почти все

te32012 committed:
Merge branch 'deployment' into db-new-code
